src/transformStateWithClones.js

```diff
@@ -7,7 +7,54 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+  let changeState = stateCopy;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        {
+          const addState = Object.assign({}, changeState, action.extraData);
+
+          result.push(addState);
+          changeState = addState;
+        }
+        break;
+
+      case 'removeProperties':
```

Comment:
For "removeProperties", you can simply delete the keys in the copy instead of creating a new object for each change

Reply:
Got it, thanks